vendor/bat-native-ledger/src/bat/ledger/internal/endpoint/promotion/post_suggestions_claim/post_suggestions_claim.cc

```diff
@@ -73,43 +72,73 @@ type::Result PostSuggestionsClaim::CheckStatusCode(const int status_code) {
   return type::Result::LEDGER_OK;
 }
 
+mojo::StructPtr<type::UrlRequest> PostSuggestionsClaim::GetSuggestionRequest(
+    const type::BraveWalletPtr wallet,
+    const char* path,
+    const std::string& payload) {
+  auto headers = util::BuildSignHeaders(path, payload, wallet->payment_id,
+                                        wallet->recovery_seed);
+
+  auto request = type::UrlRequest::New();
+  request->url = GetUrl();
+  request->content = payload;
+  request->headers = headers;
+  request->content_type = "application/json; charset=utf-8";
+  request->method = type::UrlMethod::POST;
+  return request;
+}
+
 void PostSuggestionsClaim::Request(
     const credential::CredentialsRedeem& redeem,
     PostSuggestionsClaimCallback callback) {
-  auto url_callback = std::bind(&PostSuggestionsClaim::OnRequest,
-      this,
-      _1,
-      callback);
+  auto url_callback =
+      std::bind(&PostSuggestionsClaim::OnRequest, this, _1, callback);
 
   const std::string payload = GeneratePayload(redeem);
 
-  const auto wallet = ledger_->wallet()->GetWallet();
+  auto wallet = ledger_->wallet()->GetWallet();
   if (!wallet) {
     BLOG(0, "Wallet is null");
-    callback(type::Result::LEDGER_ERROR);
+    callback(type::Result::LEDGER_ERROR, "");
     return;
   }
 
-  auto headers = util::BuildSignHeaders(
-      "post /v1/suggestions/claim",
-      payload,
-      wallet->payment_id,
-      wallet->recovery_seed);
-
-  auto request = type::UrlRequest::New();
-  request->url = GetUrl();
-  request->content = payload;
-  request->headers = headers;
-  request->content_type = "application/json; charset=utf-8";
-  request->method = type::UrlMethod::POST;
+  auto request = GetSuggestionRequest(std::move(wallet),
+                                      "post v2/suggestions/claim", payload);
   ledger_->LoadURL(std::move(request), url_callback);
 }
 
 void PostSuggestionsClaim::OnRequest(
     const type::UrlResponse& response,
     PostSuggestionsClaimCallback callback) {
   ledger::LogUrlResponse(__func__, response);
-  callback(CheckStatusCode(response.status_code));
+  auto result = CheckStatusCode(response.status_code);
+  if (result != type::Result::LEDGER_OK) {
+    callback(result, "");
+    return;
+  }
+
+  base::Optional<base::Value> value = base::JSONReader::Read(response.body);
+  if (!value || !value->is_dict()) {
+    BLOG(0, "Invalid JSON");
+    callback(type::Result::RETRY, "");
```

Comment:
I don't think that callers use the `RETRY` result type, so maybe let's just use `type::Result::LEDGER_ERROR` here and below?